### ISSUE TITLE
Completion of scala keywords

### ISSUE THREAD

**Issue by tbje** [ensime/ensime-server#1009]
Scala keywords like implicit and override does not show up in the completion list, would be nice if it did.

**Comment by fommil** [comment]
For emacs, I recommend using the `company-code` backend to company mode.

FWIW, I'm not keen on keyword completions because they get in the way, so I do this

https://github.com/fommil/dotfiles/blob/master/.emacs.d/init.el#L489

```elisp
            (set (make-local-variable 'company-backends)
                 '(ensime-company (company-keywords company-etags)))
```

which means "use `ensime-company` when ENSIME is up, otherwise fallback to `keywords` and `etags`).

You'd want something like this

```elisp
            (set (make-local-variable 'company-backends)
                 '((ensime-company company-keywords) (company-keywords company-etags)))
```

**Comment by fommil** [comment]
hmm, that setup does have the weird side effect that when ENSIME is down, and you type something that looks like a keyword, then you won't get etags completions (because company-mode only uses the first completer group that says it can do a completion)

**Comment by fommil** [comment]
Hmm, I suppose that doesn't help if you're not on emacs. Maybe the server should add keyword completions to its list.

**Comment by rorygraves** [comment]
Fixed in #1011
